### PR TITLE
Performance Improvements (Split into Core + Intg. SDKs)

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

> Main aim of the changes made is to separate core SDK from the integrations and load the necessary SDKs during runtime. The integrations needed are identified from the source config data. Thereby, reducing the overall SDK size which allows faster load times on the client side.

- Integration names identified from source config are used to asynchronously load corresponding SDKs.
  - When an integration SDK is loaded, the underlying native integration SDK load is triggered.
  - After all the integration SDK instances report initialisation complete status (timeout), cached events are replayed.
  - Added 'cdnBaseURL' to options in load API call.
  - The base CDN URL for loading the integration SDKs is determined from (in the order of precedence) load options, script tag of the core SDK and Rudder Stack CDN.
- Updated all the integration exports to named.
- Move code around and updated few methods in storage.js.
- Typos corrected.
- Re-factored certain part of the code to avoid redundancy.
- Commented out logger 'debug' statements.
- Unused code deleted.
- eslint errors/warnings addressed.
- Unused files have been moved to corresponding 'archive' directories to reduce clutter.
- Deprecated Autotrack feature removed.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/321)

### CODE REVIEW

**Review by psrikanth88 — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rudder-analytics",
-  "version": "1.2.8",
+  "version": "2.0.0",
```
Comment (rivajunior) [inline package.json:3]: I think you should not change this before it is meged. And also, if it have to change a major version it means that that PR have breaking changes. So on the description of the PR you should mark the breaking change option

Comment (saikumarrs) [reply package.json:3]: @rivajunior Although these are not breaking changes, deviating from the general convention, we've changed the major version to indicate the big shift in our SDK functionality. Anyway, thanks for pointing it out and we hope you understand.

File: README.md [hunk 2]
```diff
@@ -58,7 +59,7 @@ To load `rudder-analytics.js` on to your page synchronously, you can refer to th
 
 ```html
 <script>
-rudderanalytics=window.rudderanalytics=[];for(var methods=["load","page","track","identify","alias","group","ready","reset","getAnonymousId","setAnonymousId"],i=0;i<methods.length;i++){var method=methods[i];rudderanalytics[method]=function(a){return function(){rudderanalytics.push([a].concat(Array.prototype.slice.call(arguments)))}}(method)}rudderanalytics.load(<WRITE_KEY>,<DATA_PLANE_URL>),rudderanalytics.page();
+  rudderanalytics=window.rudderanalytics=[];for(var methods=["load","page","track","identify","alias","group","ready","reset","getAnonymousId","setAnonymousId"],i=0;i<methods.length;i++){var method=methods[i];rudderanalytics[method]=function(a){return function(){rudderanalytics.push([a].concat(Array.prototype.slice.call(arguments)))}}(method)}rudderanalytics.load(<WRITE_KEY>,<DATA_PLANE_URL>),rudderanalytics.page();
```
Comment (saikumarrs) [inline README.md:62]: @MoumitaM I don't think this change is needed

File: buildspec.staging.yaml [hunk 3]
```diff
@@ -13,10 +13,8 @@ phases:
       - npm run buildProdBrowser
       - sed 's|//# sourceMappingURL=rudder-analytics.min.js.map||' dist/rudder-analytics.min.js  > dist/prod.js
       - mv dist/prod.js dist/rudder-analytics.min.js
-      - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/rudder-analytics-staging.min.js --cache-control max-age=3600 --acl public-read
-      - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/v1/rudder-analytics-staging.min.js --cache-control max-age=3600 --acl public-read
-      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/v1/rudder-analytics-staging.min.js"
-      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/rudder-analytics-staging.min.js"     
+      - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/v3/rudder-analytics-staging.min.js --cache-control max-age=3600 --acl public-read
+      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/v3/rudder-analytics-staging.min.js"
```
Comment (saikumarrs) [inline buildspec.staging.yaml:17]: @MoumitaM please fix the paths here

File: integrations/GoogleAds/index.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
-import { GoogleAdsNode } from "./node";
```
Comment (saikumarrs) [inline integrations/GoogleAds/index.js:1]: @MoumitaM this seems to be mistake. Let's correct it.

CC @ruchiramoitra

Comment (saikumarrs) [reply integrations/GoogleAds/index.js:1]: @MoumitaM the same mistake is repeated in other destination below.

**Pull request merged**